agent/structs/config_entry_gateways.go

```diff
@@ -920,6 +924,35 @@ func (e *BoundAPIGatewayConfigEntry) Normalize() error {
 }
 
 func (e *BoundAPIGatewayConfigEntry) Validate() error {
+	allowedCertificateKinds := map[string]bool{
+		InlineCertificate: true,
+	}
+	allowedRouteKinds := map[string]bool{
+		HTTPRoute: true,
+		TCPRoute:  true,
+	}
+
+	// These should already be validated by upstream validation
+	// logic in the gateways/routes, but just in case we validate
+	// here as well.
+	for _, listener := range e.Listeners {
+		for _, certificate := range listener.Certificates {
+			if !allowedCertificateKinds[certificate.Kind] {
+				return fmt.Errorf("unsupported certificate kind: %q", certificate.Kind)
```

Comment:
Similar to #15855 , I think it'd be helpful to list the acceptable kinds in the error message here and for routes below